### ISSUE TITLE
Failure to run UtilPack.NuGet.MSBuild.NuGetTaskRunnerFactory

### ISSUE THREAD

**Issue by Sumo-MBryant** [stazz/UtilPack#1]
When attempting to build the following project using .NET Core SDK 2.0 it fails due to what appears to be missing function implementations in an updated interface:

Project:

```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <VersionPrefix>1.0.0</VersionPrefix>
    <TargetFramework>netstandard1.6</TargetFramework>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="UtilPack.NuGet.MSBuild" Version="1.0.1"/>
  </ItemGroup>

  <UsingTask
    Condition=" '$(UtilPackNuGetMSBuildAssemblyPath)' != '' "
    TaskFactory="UtilPack.NuGet.MSBuild.NuGetTaskRunnerFactory"
    AssemblyFile="$(UtilPackNuGetMSBuildAssemblyPath)"
    TaskName="UtilPack.NuGet.Push.MSBuild.PushTask">
    <Task>
      <NuGetTaskInfo>
        <PackageID>UtilPack.NuGet.Push.MSBuild</PackageID>
        <PackageVersion>1.0.0</PackageVersion>
      </NuGetTaskInfo>
    </Task>
  </UsingTask>

  <Target Name="PushTest" AfterTargets="Pack">
    <UtilPack.NuGet.Push.MSBuild.PushTask
      PackageFilePath="$(PackageOutputPath)/$(AssemblyName).$(VersionPrefix).nupkg"
    />
  </Target>

</Project>
```

```bash
$ dotnet --version
2.0.0
```

```bash
$ dotnet msbuild
Microsoft (R) Build Engine version 15.3.409.57025 for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.
```

```bash
$ dotnet msbuild /target:Pack
Microsoft (R) Build Engine version 15.3.409.57025 for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  Test -> bin\Debug\netstandard1.6\Test.dll
  Successfully created package 'bin\Debug\Test.1.0.0.nupkg'.
Test.csproj(26,5): error MSB4175: The task factory
"UtilPack.NuGet.MSBuild.NuGetTaskRunnerFactory" could not be loaded from the assembly
"~\.nuget\packages\utilpack.nuget.msbuild\1.0.1\build\\netcoreapp1.1\UtilPack.NuGet.MSBuild.dll".
Method 'Log' in type 'UtilPack.NuGet.Common.MSBuild.NuGetMSBuildLogger' from assembly
'UtilPack.NuGet.MSBuild, Version=1.0.0.0, Culture=neutral, PublicKeyToken=6b85dda6d85690de'
does not have an implementation.
```

It appears that `ILogger` was changed to include some new methods:

https://github.com/NuGet/NuGet.Client/blame/dev/src/NuGet.Core/NuGet.Common/Logging/ILogger.cs

**Comment by stazz** [comment]
Hi @Sumo-MBryant!
Thank you for reporting issue and pointing to the correct cause.

I have successfully reproduced this issue - it seems that it only happens in .NET Core MSBuild, the .NET Desktop MSBuild works fine.
It seems that if anything tries to load NuGet assemblies in .NET Core version of MSBuild, it will always use the bundled (4.3.0) version.
I ran the `dotnet` with tracing on (`COREHOST_TRACE` environment variable set to `1`), and it indicated that e.g. `NuGet.Common.dll` assembly is part of trusted platform assembly set, and is located in `sdk/2.0.0/NuGet.Common.dll` within `dotnet` installation folder (on Windows).
The NuGet assemblies are part of trusted platform assembly set because `dotnet` was given `msbuild` command.
Therefore, the workaround of separately invoking `PushTest` task will not work either, since the runtime will _always_ load NuGet assemblies from `sdk/2.0.0` folder instead of the folder with task factory assembly (`~\.nuget\packages\utilpack.nuget.msbuild\1.0.1\build\\netcoreapp1.1\`).

Since there is no workaround I am aware of, I will fix this issue asap.
The fix will require new release to be made, but I hope it will be ready tomorrow.
Most likely, I will switch all UtilPack.NuGet.* projects to use version 4.3.0, since that is easiest and most straightforward solution, and 4.3.0 does not raise the minimum platform requirements compared to 4.0.0.
